telemetry_package/main.py

```diff
@@ -1,28 +1,74 @@
-import sys
+import json
 import rclpy
-from rclpy.node import Node
+import socket
+import os
+import select
 
+from rcl_interfaces.msg import Log
 from common_package_py.common_node import CommonNode
 
+DEFAULT_UNIX_SOCKET_PATH = "/tmp/thi_drone"
 
 class TelemetryNode(CommonNode):
-    """A class that represents a telemetry node.
-    """
 
-    def __init__(self, id: str):
-        """Creates a new TelemetryNode.
+    def __init__(self):
+        # The creation of the needed sockets takes place in the constructor
+        super().__init__('telemetry_node')
 
-        Args:
-            id (str): Unique node id
-        """
+        if os.path.exists(DEFAULT_UNIX_SOCKET_PATH): 
+            print('Socket already exists')
+            raise FileExistsError
+            
+        # Create server socket
+        self.server_sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
 
-        super().__init__(id)
+        try:
+            # Bind server socket and listen for a connection. The node is blocked as long as a connection does not exist
+            self.server_sock.bind(DEFAULT_UNIX_SOCKET_PATH)
+            self.server_sock.listen(1)
+
+            # Getting the reference for the client socket
+            readable, _, _, = select.select([self.server_sock], [], [])
+            for sock in readable:
+                if sock == self.server_sock:
+                    # Accept the connection
+                    self.client_sock, _ = self.server_sock.accept()
+                    print('connection accepted. Creating Subscription')
+
+                    '''
+                    Creating the subscription after the connection has accepted ensures that the callback function is executed only after the soccket is connected
+                    '''
+                    self.subscription = self.create_subscription(
+                        Log,
+                        '/rosout',
+                        self.subscription_callback, 
+                        100)
+                    self.subscription
+
+        # In case of exceptions, close the socket and remove its path from disk
+        except Exception as e:
+            print(f'Error occurred: {e}')
+            self.server_sock.close()
+            os.remove(DEFAULT_UNIX_SOCKET_PATH)
+        
+ 
+    def subscription_callback(self, log_msg):
+        try: 
+            '''
+            Apparently, some characters in the log messages upset something in the json.loads(<received_data>.decode()) function. This is just a brute sanitization of the string, which needs to be implemented in a function of its own
+            '''
+            log_msg.msg = log_msg.msg.replace(':', '').replace('"', '').replace("'", "").replace("\{", "").replace("}", "")
+
+            self.client_sock.sendall(json.dumps({"type": "std" , "content" :  log_msg.msg}).encode())
```

Comment:
Can you please provide me with examples of messages that were problematic?
You're doing a lot of sanitizing here, which on one hand will make literally all my log messages unreadable (e.g. `:`) and on the other hand looks why too much for me.

Reply:
I am currently tryingt to figure out which messages cause this problem. Currently, when I run the node for testing, the error gets output by the listener program provided by @di-math when the program receives and tries to decode two messages right behind the other. I cannot really explain it that well yet, but I found consistent behavior using print statements in the terminal. Screenshot for reference: 

![image](https://github.com/THI-Drone/telemetry_package/assets/166122914/d45d6d85-ff96-46a5-8599-a0d1bcdb08d5)

![image](https://github.com/THI-Drone/telemetry_package/assets/166122914/f04c774b-7aac-4a03-97d9-f18e0ba8f8ba)

As you can see, the error gets output only when the receiver gets two messages on the same line (I'm sorry for the lack of a better explenation of the problem). I'll  be looking deeper into detail to understand why that's the case. I might be wrong and the examples I provided in the screenshots above are just due to some weird formatation in the terminal.

Reply:
I can tell you why your code is falling apart: You have two JSONs stitched together. The result is an invalid JSON which obviously breaks.